### PR TITLE
Add copy-file-on-save recipe

### PR DESCRIPTION
**Pull request by zonuexe**
### Brief summary of what the package does

Auto deployment (copy) a file when the file saved.

### Direct link to the package repository

https://github.com/zonuexe/emacs-auto-deployment

### Your association with the package

I'm maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: recipes/auto-deployment [hunk 1]
```diff
@@ -0,0 +1 @@
+(composer :fetcher github :repo "zonuexe/emacs-auto-deployment")
```
Comment (tarsius) [inline recipes/auto-deployment]: The first element has to match the name of the package.

**Comment by alphapapa** [comment]
Does this provide any benefit over, e.g. https://github.com/wasamasa/firestarter?  It even has an example of deployment:

```
# -*- firestarter: "rsync -avz -e ssh /src host:/dest" -*-
```

File: recipes/auto-deployment (hunk 1 above)
Comment (zonuexe) [inline recipes/auto-deployment]: @tarsius Thanks. It fixed.

**Comment by zonuexe** [comment]
@alphapapa 
Firestarter is a nice product, but the aim of auto-deployment is different from that.
Auto-deployment focuses on synchronizing saved edited file.  This can possibly be realized with firestarter, but it will be slightly more complicated.

**Comment by purcell** [comment]
Thanks. You'll need to either declare a package dependency on `cl-lib`, or on a newer Emacs version: `package-lint` should have warned you about that...?

It's not super clear from the `.el` file and its docstrings what the package does, and why a user might need it. The deployment method looks configurable, but only `copy-file` really does anything useful: setting the deployment method to a function would be pretty much the same as adding a custom function directly to `after-save-hook`.

My feeling is that you should simplify this slightly, and instead call it `copy-file-on-save`, which would also make its purpose clearer.

**Comment by zonuexe** [comment]
@purcell 
I applied simplify and renamed to `copy-file-on-save`.  Please review this again.

**Comment by purcell** [comment]
Looks great, thanks!

**Comment by zonuexe** [comment]
Thanks!

**Pull request merged**